### PR TITLE
[RFC] Get rid of broken 'named templateselect' functionality. 🔥

### LINKED ISSUE #5068: [RFC] Consolidate templatefields and template names in theme.yml
Currently we have two different objects in `theme.yml` for adding metadata to template files (names and custom fields)

```yml
templatefields:
    extrafields.twig:
        section_1:
            type: text
            label: Section 1
        section_2:
            type: html
            label: Section 2
        image:
            type: image
    anotherextrafields.twig:
        markdown:
            type: markdown
            label: A markdown field
        geolocation:
            type: geolocation
            label: A geolocation field
        image:
            type: image

# There can be defined the templates for the template chooser
templateselect:
    templates:
         - name: 'Template with extra fields'
           filename: 'extrafields.twig'
         - name: 'Another template with extra fields'
           filename: 'anotherextrafields.twig'
```

This isn't particularly DRY (what if we change the filename of a template?) and will lead to more issues if we come up with more metadata around templates. I think we should bundle them like so:

```yml
templates:
    record.twig:
        name: Default Template
        fields:
            # Insert fields here
```

Details
-------

 - Relevant Bolt Version: 2.2/master

### LINKED ISSUE #5119: Records can't select templates
image

### PR DESCRIPTION
**Pull request by bobdenotter**
Fixes #5119. Closes #5068

### CODE REVIEW

**Comment by bobdenotter** [comment]
Also fixes #4577

**Comment by GawainLynch** [comment]
WFM... Someone with a properly working solution can resubmit :wink:

**Comment by bobdenotter** [comment]
We're going to vote on this during dev-meeting tonight. :-)

**Comment by GawainLynch** [comment]
Well, it causes data loss, so the "feature" is invalid as-is. The only thing we need to vote on is the person that presses the merge button on this PR.

File: src/Twig/Handler/RecordHandler.php [hunk 1]
```diff
@@ -178,21 +178,8 @@ public function listTemplates($filter = null, $safe = false)
...
 
-        // Check: Are the templates for template chooser defined?
-        if ($themeConfig) {
-            foreach ($themeConfig as $templateFile) {
-                if (!empty($templateFile['name']) && !empty($templateFile['filename'])) {
-                    $files[$templateFile['filename']] = $templateFile['name'];
-                }
-            }
-
-            return $files;
```
Comment (bobdenotter) [inline src/Twig/Handler/RecordHandler.php]: Seriously?! If it returns here, it's no wonder the other options are missing. :fire:

**Comment by GawainLynch** [comment]
#5127 wins… good news is, it is :beers: time @bobdenotter

**Pull request closed without merging**